main.cpp

```diff
@@ -131,13 +130,13 @@ int main()
     }
     else
     {
-        std::cout << "Error! The compare function has returned a nullptr due to the result being equal values. This could change further if alternative conditionals are implemented into the function." << std::endl;
+        std::cout << "Error! The compare function returns a nullptr if one or both of the input pointers are null." << std::endl;
```

Comment:
This is wrong.
Also, the compare function's arguments can be modified into references as per the instructions.
- [ ] revise
- [ ] revise the compare function